### PR TITLE
Add dockerfile to build the image

### PR DESCRIPTION
**Pull request by mmorhun**
Adds dockerfile to build CA certificate generation job image and related resources.

Signed-off-by: Mykola Morhun <mmorhun@redhat.com>

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM alpine
```
Comment (benoitf) [inline Dockerfile]: FYI you're missing copyright headers

Also alpine version needs to be specified like `FROM alpine:<tag>`

File: build.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
```
Comment (benoitf) [inline build.sh:1]: FYI copyright missing

File: entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
```
Comment (benoitf) [inline entrypoint.sh:1]: FYI copyright headers

File: entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+CA_KEY_FILE='ca.key'
+CA_CERT_FILE='ca.crt'
+
+# Generate private key for root CA
+# Options:
+#  -out : name of file to write generated key to
+#  4096 : number of bits in the key
+openssl genrsa -out $CA_KEY_FILE 4096
```
Comment (benoitf) [inline entrypoint.sh:17]: Hello,

here we build cert at runtime (every time we launch the container)
Is it ok ?

Comment (mmorhun) [reply entrypoint.sh:17]: Hello. Yes it is exactly for what this image is designed. It should be run as Kubernetes job which generates CA private key and certificate.

Comment (benoitf) [reply entrypoint.sh:17]: ok 👍

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+
+RUN apk add --no-cache openssl curl && \
+    cd /usr/local/bin && \
+    curl -s -LO https://storage.googleapis.com/kubernetes-release/release/`curl -s https://storage.googleapis.com/kubernetes-release/release/stable.txt`/bin/linux/amd64/kubectl && \
```
Comment (benoitf) [inline Dockerfile]: probably need to use a fixed version of kubernetes release 
and ENV variable with K8S version

to be sure that everytime we build the container we have the same behavior

File: build.sh [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+
+TAG=$(git rev-parse --short HEAD)
+
+docker build -t quay.io/eclipse/che-cert-manager-ca-cert-generator:${TAG} .
```
Comment (benoitf) [inline build.sh:12]: You may provide a .github action as well to automatically build and publish the image on every commit and check build on PR ?

File: Dockerfile (hunk 1 above)
Comment (mmorhun) [reply Dockerfile]: ok

File: Dockerfile (hunk 5 above)
Comment (mmorhun) [reply Dockerfile]: Agree, thanks.

File: build.sh (hunk 6 above)
Comment (mmorhun) [reply build.sh:12]: Sound interesting! But I would like to have it in a separate PR.

File: entrypoint.sh [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+# Generate CA certificate and sign it with previously generated key.
+# Options:
+#  -batch : script (non-interactive) mode
+#  -new : creates new sertificate request
+#  -x509 : produces self signed sertificate instead of certificate request
+#  -deys : number of days this certificate will be valid for
```
Comment (AndrienkoAleksandr) [inline entrypoint.sh]: deys -> days

Comment (mmorhun) [reply entrypoint.sh]: Thanks

**Review by AndrienkoAleksandr — APPROVED** [review]

**Review by benoitf — APPROVED** [review]

**Pull request merged**